src/basics.php

```diff
@@ -143,6 +143,15 @@ function dd($var, $showHtml = null): void
             'file' => $trace[0]['file'],
         ];
 
+        if (PHP_SAPI !== 'cli') {
+            $referer = env('HTTP_REFERER') ?? '';
```

Comment:
$referer = (string)env('HTTP_REFERER');

suffices

Reply:
next pr, i make from ide, worrying about this.